src/data/fintype/basic.lean

```diff
@@ -229,6 +229,12 @@ def of_surjective [decidable_eq β] [fintype α] (f : α → β) (H : function.s
   fintype β :=
 ⟨univ.image f, λ b, let ⟨a, e⟩ := H b in e ▸ mem_image_of_mem _ (mem_univ _)⟩
 
+instance decidable_subtype_of_fintype {α : Sort*} [fintype α] (p : α → Prop) [decidable_pred p] :
```

Comment:
I think you should call this `subtype_of_fintype` and simply remove the other instance - if people don't care about computability, then they can use this instance with classical.decidable

Reply:
I'm afraid that will break a ton of things. I'll check.

Reply:
Last time I removed a no computable instance it actually made things less broken elsewhere.

Reply:
So many things rely on inferring `fintype ↥s` for arbitrary `set α`, where it is not clear how one proves `[decidable_pred s]`. An example would be `set.to_finset`. So I think this would be a large refactor, to insert `classical` across the whole library.

Reply:
Could you push that version to a new branch, so that I have a cache to play around with that larger refactor in later today?

Reply:
https://github.com/leanprover-community/mathlib/tree/pechersky/abolish-subtype-of-fintype

Reply:
I've made some pushes to that branch - I get the feeling that not that much needs fixing, it just takes a while to find each spot.

Reply:
In the meantime, how about replacing `subtype_of_fintype` with `by classical; exact decidable_subtype_of_fintype`? That should make the instances defeq up to decidable arguments, which hopefully makes any diamond that appear slightly less bad.

Reply:
The branch now passes CI :)

Reply:
If Eric managed to do the refactor, that approach is probably the better one. @pechersky what do you think?